config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/beans/JmsDestination.java

```diff
@@ -0,0 +1,351 @@
+/*
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+package com.ca.apim.gateway.cagatewayconfig.beans;
+
+import com.ca.apim.gateway.cagatewayconfig.config.loader.ConfigLoadException;
+import com.ca.apim.gateway.cagatewayconfig.config.spec.ConfigurationFile;
+import com.ca.apim.gateway.cagatewayconfig.config.spec.EnvironmentType;
+import com.ca.apim.gateway.cagatewayconfig.util.IdGenerator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import org.jetbrains.annotations.Nullable;
+
+import javax.inject.Named;
+
+import java.io.File;
+import java.util.Map;
+
+import static com.ca.apim.gateway.cagatewayconfig.config.spec.ConfigurationFile.FileType.JSON_YAML;
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+import static java.util.Arrays.stream;
+
+@JsonInclude(NON_NULL)
+@Named("JMS_ENDPOINT")
+@ConfigurationFile(name = "jms-destinations", type= JSON_YAML)
+@EnvironmentType("JMS_DESTINATION")
+@SuppressWarnings("squid:S2068") // sonarcloud believes this is a hardcoded password
+public class JmsDestination extends GatewayEntity {
+    
+    // Basics
+    private String providerType;
+    
+    // JNDI
+    private String initialContextFactoryClassName;
+    private String jndiUrl;
+    private String jndiUsername;
+    private String jndiPasswordRef;
+    private String jndiPassword;
+    private Map<String, Object> jndiProperties;
+
+    // Destination
+    private DestinationType destinationType;
+    private String connectionFactoryName;
+    private String destinationName;
+    private String destinationUsername;
+    private String destinationPasswordRef;
+    private String destinationPassword;
+    
+    // Inbound
+    private InboundJmsDestinationDetail inboundDetail;
+    
+    // Outbound
+    private OutboundJmsDestinationDetail outboundDetail;
+    
+    // (kpak) - add support for providers with more configuration settings:
+    // TIBCO EMS
+    // WebSphere MQ over LDAP
+    // private Map<String, Object> additionalProperties;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This block of commented-out lines of code should be removed. ([squid:CommentedOutCodeLine](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3ACommentedOutCodeLine))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=138&issues=AWekObW6bde-piqdGWhM&open=AWekObW6bde-piqdGWhM)

[ISSUE_KEY]: AWekObW6bde-piqdGWhM